hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -337,9 +337,15 @@ private void refreshTimeline() throws IOException {
     }
 
     JavaRDD<GenericRecord> avroRDD = avroRDDOptional.get();
+    if (writeClient == null) {
+      this.schemaProvider = schemaProvider;
+      setupWriteClient();
```

Comment:
all we need is a config here. don't think we need to initialize writeClient here.

Reply:
also thinking do we really need to instantiate the config. since it is just one property, can't we directly read if from TypedProperties? @bvaradar : do you have any thoughts on this. Basically we need to read just one config value for deleteMarker from the properties set. This step is little ahead of where we instantiate writeClient, so wondering how to go about it.

Reply:
+1, lets not create the write client here. As we are setting the field (including defaults) from DataSourceReadOptions, you can use the TypedProperties to get it.